roles/etcd/templates/etcd.env.j2

```diff
@@ -32,8 +32,9 @@ ETCD_MAX_SNAPSHOTS={{ etcd_max_snapshots }}
 {% if etcd_max_wals is defined %}
 ETCD_MAX_WALS={{ etcd_max_wals }}
 {% endif %}
-# Flannel need etcd v2 API
+{% if hostvars[groups['k8s_cluster'][0]]['kube_network_plugin'] == 'flannel' %}
```

Comment:
btw why not simple using `kube_network_plugin == 'flannel'` ?

Reply:
In the example inventory, the variable is defined in `group_vars/k8s_cluster/k8s-cluster.yml` ; so if we use a separate etcd, it will not be defined for the etcd nodes.